hypertrace-gradle-publish-plugin/src/main/java/org/hypertrace/gradle/publishing/PublishPlugin.java

```diff
@@ -1,217 +1,134 @@
-                  publication -> publication.from(project.getComponents().getByName("java")));
-            });
-  }
-
-  private PublishingExtension getPublishingExtension() {
-    return project.getExtensions().getByType(PublishingExtension.class);
-  }
-
-  private boolean isJavaGradlePluginPluginApplied() {
-    return project.getPluginManager().hasPlugin("java-gradle-plugin");
+    project.getPluginManager()
+      .withPlugin("java-library", appliedPlugin -> {
+        if (this.isJavaGradlePluginPluginApplied()) {
+          return; // This already creates a publication, we don't want to duplicate
+        }
+        publications.create("javaLibrary", MavenPublication.class, publication -> {
+          publication.from(project.getComponents().getByName("java"));
+          updatePomMetadata(publication);
+        });
+      });
   }
 
-  private void configureUploadTask(BintrayUploadTask task) {
-    this.validateExtensionAtConfigurationTime();
-    task.doFirst(unused -> this.validateExtensionAtExecutionTime());
-    Project project = task.getProject();
-    task.project = project; // Why do they have their own var?
-    task.setUser(this.extension.user.getOrNull());
-    task.setApiKey(this.extension.apiKey.getOrNull());
-    task.setApiUrl(this.extension.apiUrl.get());
-    task.setPublish(true);
-    task.setOverride(false);
-    task.setRepoName(this.extension.repo.get());
-    task.setPackageName(this.extension.name.get());
-    task.setPackageVcsUrl(this.extension.vcsUrl.get());
-    task.setUserOrg(this.extension.organization.get());
-    if (this.extension.license.get() != TRACEABLE_COMMUNITY) {
-      // The bintray plugin doesn't support custom licenses. This is the default on all repos.
-      task.setPackageLicenses(this.extension.license.get().bintrayString);
-    }
-    task.setVersionName(String.valueOf(project.getVersion()));
+  private void addDistributionPublicationWhenApplied(PublicationContainer publications) {
+    project.getPluginManager()
+      .withPlugin("distribution", appliedPlugin ->
+        publications.create("distributionZip", MavenPublication.class, publication ->
+          publication.artifact(project.getTasks()
+            .getByName("distZip"))));
   }
 
-  private Collection<Task> getDependenciesForPublication(Project project, Publication publication) {
-    if (!(publication instanceof MavenPublication)) {
-      return Collections.emptySet();
-    }
-    return project.getTasksByName(this.getPublishLocalTaskNameForPublication(publication), false);
-  }
+  private void updatePomMetadata(MavenPublication publication) {
+    publication.pom(mavenPom -> {
+      // Add url
+      mavenPom.getUrl().set(extension.url);
 
-  private String getPublishLocalTaskNameForPublication(Publication publication) {
-    return String.format("publish%sPublicationToMavenLocal", capitalize(publication.getName()));
+      // Add license
+      if (extension.license.isPresent()) {
```

Comment:
We say this is required, right? Where do we enforce that? We shouldn't skip silently (although at this point in execution, not sure if the build script has had a chance to eval yet.

Reply:
added check for license field.